### DISCUSSION TITLE
how to get 1 bit value from bit field in view

### DISCUSSION THREAD

**Discussion by flyly0755** [sqlalchemy/sqlalchemy#7025] [Usage Questions] [answered]
In the table  I have one bit field with 8 bits, and now i want to create a corresponding view to expand as 8 field.
code like below
```python
from sqlalchemy.dialects.postgresql import BIT
from sqlalchemy import select
class TbTest(Base):
    __tablename__ = 'tb_test'
    bitfield = Column(BIT(length=8), nullable=False)  # B*1
```
stmtStdfPtr = select([
    (TbTest.bitfield.get_bit(1).label('bit1'),
    (TbTest.bitfield.get_bit(2).label('bit2'),
     ...
like the select language in postgresql:
select test3,get_bit(test3,0) as getbit7,get_bit(test3,1) as getbit6,get_bit(test3,2) as getbit5,get_bit(test3,3) as getbit4,
get_bit(test3,4) as getbit3,get_bit(test3,5)as getbit2,get_bit(test3,6)as getbit1,get_bit(test3,7)as getbit0
from test_table;

**Comment by CaselIT** [comment, marked as answer]
Hi,

Using `func` does not work? something like `sa.func.get_bit(TbTest.bitfield, 1).label('bit1')` etc

**Reply by flyly0755** [reply]
Thank you soooooooo much, that is exactly what i want!!!!!!!

**Reply by CaselIT** [reply]
Nice to hear. As mentioned in the docs https://docs.sqlalchemy.org/en/14/core/functions.html func can be used to render any function, even user generated procedures that have custom names
